Update HarpoonType.java

vi3tcn opened the pull request:
Added all the other harpoon types and their ornament kits:

DRAGON_HARPOON_OR(ItemID.DRAGON_HARPOON_OR, AnimationID.FISHING_TRAILBLAZER_HARPOON, "Dragon harpoon (or)"),

DRAGON_HARPOON_OR_30349(ItemID.DRAGON_HARPOON_OR_30349, AnimationID.FISHING_TRAILBLAZER_HARPOON, "Dragon harpoon reloaded (or)"),  INFERNAL_HARPOON_UNCHARGED(ItemID.INFERNAL_HARPOON_UNCHARGED, AnimationID.FISHING_INFERNAL_HARPOON, "Infernal harpoon (uncharged)"),

INFERNAL_HARPOON_OR_30342(Item.ID.INFERNAL_HARPOON_OR_30342, AnimationID.FISHING_TRAILBLAZER_HARPOON, "Infernal harpoon (or)"),

INFERNAL_HARPOON_UNCHARGED_30343(INFERNAL_HARPOON_UNCHARGED_30343, AnimationID.FISHING_TRAILBLAZER_HARPOON, "Infernal harpoon uncharged (or)", CRYSTAL_HARPOON_INACTIVE(ItemID.CRYSTAL_HARPOON_INACTIVE, AnimationID.FISHING_CRYSTAL_HARPOON, "Crystal harpoon (inactive)");

Commit:
Update HarpoonType.java

Added all the other harpoon types and their ornament kits:

DRAGON_HARPOON_OR(ItemID.DRAGON_HARPOON_OR, AnimationID.FISHING_TRAILBLAZER_HARPOON, "Dragon harpoon (or)"),

DRAGON_HARPOON_OR_30349(ItemID.DRAGON_HARPOON_OR_30349, AnimationID.FISHING_TRAILBLAZER_HARPOON, "Dragon harpoon reloaded (or)"), 
INFERNAL_HARPOON_UNCHARGED(ItemID.INFERNAL_HARPOON_UNCHARGED, AnimationID.FISHING_INFERNAL_HARPOON, "Infernal harpoon (uncharged)"),

INFERNAL_HARPOON_OR_30342(Item.ID.INFERNAL_HARPOON_OR_30342, AnimationID.FISHING_TRAILBLAZER_HARPOON, "Infernal harpoon (or)"),

INFERNAL_HARPOON_UNCHARGED_30343(INFERNAL_HARPOON_UNCHARGED_30343, AnimationID.FISHING_TRAILBLAZER_HARPOON, "Infernal harpoon uncharged (or)",
CRYSTAL_HARPOON_INACTIVE(ItemID.CRYSTAL_HARPOON_INACTIVE, AnimationID.FISHING_CRYSTAL_HARPOON, "Crystal harpoon (inactive)");

Commit:
Update HarpoonType.java

Commit:
Update HarpoonType.java

Commit:
Update HarpoonType.java

Commit:
Update HarpoonType.java

edited the reloaded ornament kit names

gmason0 reviewed (changes requested):
I think it would make more sense here to use the `ItemVariationMapping` class from Runelite to determine these different item versions, such as inactive vs active - or items with ornament kits?

Commit:
Update HarpoonType.java

Edited for ItemVariationMapping so should work with all variants or ornament kits and uncharged harpoons

Commit:
Merge pull request #2 from vi3tcn/vi3tcn-patch-1

Update HarpoonType.java

gmason0 commented on runelite-client/src/main/java/net/runelite/client/plugins/microbot/tempoross/enums/HarpoonType.java:
```diff
@@ -2,34 +2,31 @@
 
 import net.runelite.api.AnimationID;
 import net.runelite.api.ItemID;
+import net.runelite.client.game.ItemVariationMapping;
```
I am not sure where this is implemented inside of the script class, but I would think it would look something like this:

```java
// HarpoonType.java - add method to create an array of item variants
public int[] getHarpoonVariations() {
	return ItemVariationMapping.getVariations(getId()).stream().mapToInt(Integer::intValue).toArray();
}
```

```java 
// In TemporossScript.java
private boolean hasHarpoon() {
	return Rs2Inventory.contains(harpoonType.getHarpoonVariations()) || Rs2Equipment.isWearing(harpoonType.getHarpoonVariations());
}
```
